packages/algo-ts/src/arc4/encoded-types.ts

```diff
@@ -237,7 +237,7 @@ type ExpandTupleType<T extends ARC4Encoded[]> = T extends [infer T1 extends ARC4
     : `${T1[typeof TypeProperty]},${ExpandTupleType<TRest>}`
   : ''
 
-export class Tuple<TTuple extends [ARC4Encoded, ...ARC4Encoded[]]> extends ARC4Encoded {
+export class Tuple<TTuple extends ARC4Encoded[]> extends ARC4Encoded {
```

Comment:
The original type constraint here is deliberate as it disallows empty arrays

Reply:
It has been reverted.